source/matplot/axes_objects/vectors.cpp

```diff
@@ -147,9 +147,14 @@ namespace matplot {
                         std::sqrt(u_value * u_value + v_value * v_value +
                                   w_value * w_value);
                     if (mag != 0.0) {
-                        u_value *= scale_ / mag;
-                        v_value *= scale_ / mag;
-                        w_value *= scale_ / mag;
+                        u_value /= mag;
+                        v_value /= mag;
+                        w_value /= mag;
+                    }
+                    if (scale_ != 0.0) {
```

Comment:
We probably don't need `if (scale_ != 0.0)`. Although this is an edge case, if the user provided `scale` as `0.0`, they should be waiting for that. Although this might be inconvenient when the user is not expecting for that, this conditional could break the behavior when it's intentional. 

For instance, this could be part of an animation where the length is varying and passing through `0.0`. The scale could even the negative in this animation. In this case, whenever the length is 0.0, the user would see a large vector blinking, which is not expected behavior. 

Besides, although we are not dealing with that in matplotplusplus yet, more and more compilers are defining an error for floating point equality comparisons.

Reply:
In `axes_type::quiver` and `axes_type::quiver3` if scale = 0, then no scaling is done. It is essentially saying turn off scaling if scale = 0 (treating it as a bool and a float). So currently if the user says scale = 0, they don't want to use scaling.

Reply:
This is in axes_type::quiver
 
        auto u_scaled =
             (scale != 0.)
                 ? transform(
                       u, [&](double u) { return (u / u_max) * scale * x_max; })
                 : u;

Reply:
I see. This was actually a workaround because there was no `normalize` yet 😆. 

You're absolutely right. Changing it would unfortunately break the current behavior. This seems like the wrong behavior but deprecating functions to change old behavior is quite problematic 🙁. 

So never mind. My bad. I'll try to remember about that when v2.0.0-alpha is out.